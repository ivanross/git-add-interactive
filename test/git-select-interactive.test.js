const {
  cleanString,
  parseFilesInfo,
  relativePath,
  getWorkingDirFilesInfo,
  getIndexFilesInfo
} = require("../utils");

//prettier-ignore
const test = [
  { path: "aa"               , index: " ", working_dir: "D" },
  { path: 'bb -> "dir/bb bb"', index: "R", working_dir: " " },
  { path: "new_file.txt"     , index: "A", working_dir: " " },
  { path: "test"             , index: "M", working_dir: " " },
  { path: "dir/cc"           , index: "?", working_dir: "?" },
  { path: "dir/test"         , index: "?", working_dir: "?" }
];

describe(cleanString, () => {
  it("should clean string", () => {
    expect(cleanString("test")).toBe("test");

    expect(cleanString('"a a"')).toBe("a\\ a");

    expect(cleanString('"src/foo/bar baz/f o o"')).toBe(
      "src/foo/bar\\ baz/f\\ o\\ o"
    );
  });
});

describe(relativePath, () => {
  it("should return path from current working dir", () => {
    expect(relativePath("/a/b/c", "/a/b/c/d", "f")).toEqual("../f");
    expect(relativePath("/a/b/c", "/a/b", "f")).toEqual("c/f");
  });
});

describe(getWorkingDirFilesInfo, () => {
  it("should work", () => {
    expect(getWorkingDirFilesInfo([])).toEqual([]);
    expect(getWorkingDirFilesInfo(test)).toMatchSnapshot();
  });
});

describe(getIndexFilesInfo, () => {
  it("should work", () => {
    expect(getIndexFilesInfo([])).toEqual([]);
    expect(getIndexFilesInfo(test)).toMatchSnapshot();
  });
});

describe(parseFilesInfo, () => {
  it("should work with index files", () => {
    const indexFiles = getIndexFilesInfo(test);
    expect(parseFilesInfo([])).toEqual([]);
    expect(parseFilesInfo(indexFiles, "", "")).toMatchSnapshot();
  });

  it("should work with working dir files", () => {
    const workingDirFiles = getWorkingDirFilesInfo(test);
    expect(parseFilesInfo([])).toEqual([]);
    expect(parseFilesInfo(workingDirFiles, "", "")).toMatchSnapshot();
  });

  it("should work with working dir files with different paths", () => {
    const workingDirFiles = getWorkingDirFilesInfo(test);
    expect(parseFilesInfo([])).toEqual([]);
    expect(
      parseFilesInfo(workingDirFiles, "/a/b/e/f", "/a/b/c/d")
    ).toMatchSnapshot();
  });
});
