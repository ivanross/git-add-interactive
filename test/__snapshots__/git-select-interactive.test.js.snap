// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getIndexFilesInfo should work 1`] = `
Array [
  Object {
    "path": "bb -> \\"dir/bb bb\\"",
    "status": "renamed",
  },
  Object {
    "path": "new_file.txt",
    "status": "new_file",
  },
  Object {
    "path": "test",
    "status": "modified",
  },
]
`;

exports[`getWorkingDirFilesInfo should work 1`] = `
Array [
  Object {
    "path": "aa",
    "status": "deleted",
  },
  Object {
    "path": "dir/cc",
    "status": "untracked",
  },
  Object {
    "path": "dir/test",
    "status": "untracked",
  },
]
`;

exports[`parseFilesInfo should work with index files 1`] = `
Array [
  Object {
    "files": Array [
      "bb",
      "dir/bb\\\\ bb",
    ],
    "label": "bb -> \\"dir/bb bb\\"",
    "status": "renamed",
    "value": "bb -> \\"dir/bb bb\\"",
  },
  Object {
    "files": Array [
      "new_file.txt",
    ],
    "label": "new_file.txt",
    "status": "new_file",
    "value": "new_file.txt",
  },
  Object {
    "files": Array [
      "test",
    ],
    "label": "test",
    "status": "modified",
    "value": "test",
  },
]
`;

exports[`parseFilesInfo should work with working dir files 1`] = `
Array [
  Object {
    "files": Array [
      "aa",
    ],
    "label": "aa",
    "status": "deleted",
    "value": "aa",
  },
  Object {
    "files": Array [
      "dir/cc",
    ],
    "label": "dir/cc",
    "status": "untracked",
    "value": "dir/cc",
  },
  Object {
    "files": Array [
      "dir/test",
    ],
    "label": "dir/test",
    "status": "untracked",
    "value": "dir/test",
  },
]
`;

exports[`parseFilesInfo should work with working dir files with different paths 1`] = `
Array [
  Object {
    "files": Array [
      "../../e/f/aa",
    ],
    "label": "../../e/f/aa",
    "status": "deleted",
    "value": "../../e/f/aa",
  },
  Object {
    "files": Array [
      "../../e/f/dir/cc",
    ],
    "label": "../../e/f/dir/cc",
    "status": "untracked",
    "value": "../../e/f/dir/cc",
  },
  Object {
    "files": Array [
      "../../e/f/dir/test",
    ],
    "label": "../../e/f/dir/test",
    "status": "untracked",
    "value": "../../e/f/dir/test",
  },
]
`;
